### ISSUE TITLE
Not possible to disable otel trace exporting at runtime

### ISSUE THREAD

**Issue by troosan** [quarkusio/quarkus#44200]
### Describe the bug

It is currently not possible to tell a Quarkus app to not export traces at runtime.
The solution proposed https://github.com/quarkusio/quarkus/issues/35306 and that is documented here https://quarkus.io/guides/opentelemetry-tracing#sampler will indeed not export traces, but the service will still try to contact the default endpoint. In my case I have an oTel collector but not yet in all environments, so I want to disable the connection to it in environments that don't have the collector available.

IMHO, if I configure an empty value for the `quarkus.otel.exporter.otlp.endpoint` property (or the trace specific one), it should not fallback to the default.
That's at least how it is documented https://quarkus.io/guides/config#inject-the-configuration, if a property is empty and defined as an Optional<xxxx>, it will be empty, instead of being set to the default value.

That's also how the impl was foreseen it seems, in the [code](https://github.com/quarkusio/quarkus/blob/a6d6cd7c017c98abf757ad69c190bfdd6a97964b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OTelExporterRecorder.java#L80) you can see that if the baseUri is null it will not try to export the traces.

Except that with the current implementation the baseUri can never be null. That's because of how [resolveTraceEndpoint()](https://github.com/quarkusio/quarkus/blob/a6d6cd7c017c98abf757ad69c190bfdd6a97964b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OTelExporterRecorder.java#L397C19-L397C39) is implemented. The current implementation will never return an empty String.

I think all we need to do is add the following, and also not default to DEFAULT_GRPC_BASE_URI, but to an empty string instead.
```java
        if (runtimeConfig.traces().endpoint().isEmpty()) {
            return "";
        }
```

The default value is handled by the properties mechanism already.

@brunobat what do you think, does this make sense? If you agree I can create a PR for this. I would then also update the documentation on how to disable properly the tracing.

The same should probably be done when resolving the endpoint for metrics (and logs as this was just added if I'm not wrong)

**Comment by troosan** [comment]
Won't this completely disable otel, also the traceId generation and propagation?
I would like to be able to just disable the exporting of the traces.

**Comment by brunobat** [comment]
Yes, it will disable everything, but you can passthrough the headers with incoming tracing data. 
No point in generating ids if they will never show up on the collector.

**Comment by troosan** [comment]
indeed, my clients are pushing their traces to a collector though, so it would be nice to at least log the traceId in the logs.
Anyway, I can indeed disable the whole thing, I just thought that the implementation didn't exactly respect the logic of the optional property and that it would be a cleaner way to disable the sending of traces than through setting the ratio to 0.

**Comment by brunobat** [comment]
Seeing the parent span and trace ids in the logs? 
That might be interesting, but it would be a new feature.
Can we close this issue?

**Comment by troosan** [comment]
it's already supported like this
[traceId=%X{traceId}, spanId=%X{spanId}]

described here: https://quarkus.io/guides/opentelemetry-tracing#create-the-configuration

**Comment by brunobat** [comment]
That's for the current span, not the parent one...
